package/composition.yaml

```diff
@@ -83,3 +89,75 @@ spec:
               string:
                 type: Convert
                 convert: FromBase64
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.manifest.data.restapi-id
+          toFieldPath: status.agw.restapi-id
+          transforms:
+            - type: string
+              string:
+                type: Convert
+                convert: FromBase64
+    - name: deployment
+      base:
+        apiVersion: apigateway.aws.upbound.io/v1beta1
+        kind: Deployment
+      patches:
+        - fromFieldPath: spec.parameters.region
+          toFieldPath: spec.forProvider.region
+        - fromFieldPath: status.agw.restapi-id
+          toFieldPath: spec.forProvider.restApiId
```

Comment:
Do we need to check that all of the methods have also fully synced? Or is the restApiId enough?

I would expect that all the methods and paths have to be created before the deployments can be created.

Does the restApiId only get patched back once fully synced?

Reply:
It relies on Connection Details here
https://github.com/ytsarev/platform-ref-apigateway/blob/extend-to-deployment-plus/package/composition.yaml#L26-L34

And Connection Details should be populated only after full Release becomes true. And Release should become ready only after all MRs are ready following `wait: true` https://github.com/upbound/platform-ref-apigateway/blob/main/package/composition.yaml#L15 

But very good point. That is something to keep additional attention to during e2e integration testing and in case any link in this chain is not reliable we can add additional ordering mechanisms 👍